resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -583,6 +584,31 @@ class ClientSuite extends SparkFunSuite with Matchers {
     }
   }
 
+  test("SPARK-35672: test Client.getUserClasspathUrls") {
+    val conf = new SparkConf()
+        .set(SECONDARY_JARS, Seq(
+          "local:/local/matching/replace/foo.jar",
+          "local:/local/not/matching/replace/foo.jar",
+          "file:/absolute/file/path/foo.jar",
+          "relative/file/path/foo.jar"
```

Comment:
we should make sure absolute paths that match the gateway root path are replaced

Reply:
I think not -- we can continue discussion on this on your other comment above. In either case, I think it's good to explicitly test for this case, so I have added two new cases:
```scala
          "/local/matching/replace/but-not-actually-local/foo.jar",
          "/absolute/path/foo.jar",
```